site/content/blog/2019-10-21-localizing-matterpoll/index.md

```diff
@@ -0,0 +1,182 @@
+---
+title: "Localizing Matterpoll"
+slug: localizing-matterpoll
+date: 2019-11-05T10:49:35+02:00
+categories:
+    - "plugins"
+    - "go"
+author: Ben Schumacher
+github: hanzei
+community: hanzei
+---
+
+[Matterpoll](https://github.com/matterpoll/matterpoll) is a plugin that allows creating polls inside of Mattermost. Given that Mattermost is [localized in 16 different languages](https://docs.mattermost.com/developer/localization.html) we surely want to accomplish the same for Matterpoll. Because we rely on contributors to do the translations we wanted to make sure it's a easy as possible for them to translate new strings and determine if already translated stings needs to be updated because the "source" text changed. On the other hand Matterpoll only has two maintainer ([@kaaka](https://github.com/kaakaa) and [me](https://github.com/hanzei)) and no own infrastructure to work with. Using some translation server like [Transifex](https://en.wikipedia.org/wiki/Transifex) or [Weblate](https://en.wikipedia.org/wiki/Weblate) is not an option.
+
+The [Mattermost Server](https://github.com/mattermost/mattermost-server) uses the [go-i18n package](https://github.com/nicksnyder/go-i18n). The library is well maintained and very popular, which gave me confidence in using it.
+
+
+#### Choosing a version
+
+When [Translations where added to Mattermost](https://github.com/mattermost/mattermost-server/commit/8e404c1dcf820cf767e9d6899e8c1efc7bb5ca96#diff-db85c0ea4d2e69c8abaefa875ba77c51) back in 2016 the latest version of go-i18 available was [`v1.4.0`](https://github.com/nicksnyder/go-i18n/releases/tag/v1.4.0). Since then the development has continued and [`v2.0.0`](https://github.com/nicksnyder/go-i18n/releases/tag/v2.0.0) was released may this year. V2 has a completely different API and also the CLI tool [`goi18n`](https://github.com/nicksnyder/go-i18n#command-goi18n-) has changed a lot. Hence, we had to make a choice: Stick with the well established V1 or use the newer V2? 
```

Comment:
```suggestion
When [translations were added to Mattermost](https://github.com/mattermost/mattermost-server/commit/8e404c1dcf820cf767e9d6899e8c1efc7bb5ca96#diff-db85c0ea4d2e69c8abaefa875ba77c51) back in 2016 the latest version of `go-i18` available was [`v1.4.0`](https://github.com/nicksnyder/go-i18n/releases/tag/v1.4.0). 

Since then the development has continued and [`v2.0.0`](https://github.com/nicksnyder/go-i18n/releases/tag/v2.0.0) was released may this year, with a completely different API. The CLI tool [`goi18n`](https://github.com/nicksnyder/go-i18n#command-goi18n-) has also changed a lot. Hence, we had to make a choice: Stick with the well-established v1 or use the newer v2? 
```

Reply:
Same questions as above